src/sinks/util/buffer/loki.rs

```diff
@@ -80,6 +124,41 @@ impl Batch for LokiBuffer {
     }
 
     fn push(&mut self, mut item: Self::Input) -> PushResult<Self::Input> {
+        // We must sort the stream labels to ensure they hash to
+        // the same stream if the label set matches.
+        item.labels.sort_unstable();
+
+        let partition = &item.partition;
+        if self.latest_timestamps.is_none() {
+            self.partition = Some(item.partition.clone());
+            self.latest_timestamps = Some(self.global_timestamps.take(&partition));
+        }
+        let latest_timestamp = self
+            .latest_timestamps
+            .as_ref()
+            .unwrap()
```

Comment:
Given that these buffers will always (?) have something pushed into them, is there any point to delayed allocation of the `latest_timestamps` map?

Reply:
Currently, the way it work is:
- Take `latest_timestamps` map of this partition from `global_timestamps` when the first event arrive.
- Put it back in `finish()`.
- New buffer is created and will take the map again.


I don't think we can remove `Option` here without changing PartitionBatchSink implementation.